src/planner/ompl/CRRTConnect.cpp

```diff
@@ -207,8 +207,12 @@ ::ompl::base::PlannerStatus CRRTConnect::solve(
     Motion* startMotion = startTree ? newmotion : lastmotion;
     Motion* goalMotion = startTree ? lastmotion : newmotion;
 
+    // NOTE: (sniyaz): Also ensure that the connection motion is collision-free.
+    bool validConnection
```

Comment:
Yeah, my concern here is that `mConnectionRadius` is an independent parameter that the user gets to set. There are cases where the user could set  `mConnectionRadius` to a large value, which would possible allow collisions (exactly as you said).

IMO this is a better solution than forcing `mConnectionRadius` to be smaller than the collision-check resolution: that's a little cumbersome since the user has to be aware of the relationship between all of these parameters, rather than handling it gracefully here.

That said, I'm willing to change if we really want: just wanted to explain my thinking behind this 😄

Reply:
The other issue with `mConnectionRadius` being too large is that the step sizes do not matter anymore at that point. I think the way to _correctly_ handle this is to force the user to choose a connection radius to be smaller than or equal to the collision checking resolution. From the docstring for connection radius, it does imply that a collision check between the two states in unnecessary since they are considered to be the same. 

If you think the user might not know the parameters correctly, how about adding a dosctring to mConnectionRadius that it needs to be <= collision resolution. If it is greater, issue an OMPL_WARN() and clamp it to the `min(mConnectionRadius, collision resolution)`.

Reply:
Agreed, that makes more sense. Done!